lib/components/SButton.vue

```diff
@@ -21,7 +22,7 @@ const props = defineProps<{
   size?: Size
   type?: Type
   mode?: Mode
-  icon?: object
+  icon?: any
```

Comment:
We can do this later, but maybe we could define this as `icon?: IconifyIcon`? And force users of the component to always pass in iconify icon (create a custom icon rather than passing in a plain SVG file if they need it).

I was wondering if we can avoid adding `v-if`s on every component receiving icon as a prop, but instead just simply do `SIcon :icon="icon` only.

Or, maybe we could make `SIcon` more abstruct and let it handle the `v-if` conditions.

Reply:
Wonderful! Thanks so much for the work 🙌